src/scraper/scraper.cpp

```diff
@@ -1245,131 +1264,383 @@ bool UserpassPopulated()
-            }
-            catch (const std::runtime_error& e)
+            ScraperFileManifestMap::iterator entry;
+            for (entry = StructScraperFileManifest.mScraperFileManifest.begin(); entry != StructScraperFileManifest.mScraperFileManifest.end(); )
             {
-                _log(logattribute::ERR, "DownloadProjectTeamFiles", "Failed to pull team header file for " + prjs.m_name);
-                continue;
-            }
+                ScraperFileManifestMap::iterator entry_copy = entry++;
 
-            if (sTeamETag.empty())
-            {
-                _log(logattribute::ERR, "DownloadProjectTeamFiles", "ETag for project is empty" + prjs.m_name);
+                if (entry_copy->second.project == prjs.m_name && entry_copy->second.current == true && entry_copy->second.filetype == "host")
+                {
+                    _log(logattribute::INFO, "DownloadProjectHostFiles", "Marking old project manifest host entry as current = false.");
+                    MarkScraperFileManifestEntryNonCurrent(entry_copy->second);
+                }
 
-                continue;
+                // If host records are older than EXPLORER_EXTENDED_FILE_RETENTION_TIME delete record, or if fScraperRetainNonCurrentFiles is false,
+                // delete all non-current records, including the one just marked non-current. (EXPLORER_EXTENDED_FILE_RETENTION_TIME rather
+                // then SCRAPER_FILE_RETENTION_TIME is used, because this section is only active if fExplorer is true.)
+                if (entry_copy->second.filetype == "host" && (((GetAdjustedTime() - entry_copy->second.timestamp) > EXPLORER_EXTENDED_FILE_RETENTION_TIME)
+                                                              || (entry_copy->second.project == prjs.m_name && entry_copy->second.current == false && !SCRAPER_RETAIN_NONCURRENT_FILES)))
+                {
+                    DeleteScraperFileManifestEntry(entry_copy->second);
+                }
             }
+
+            if (!InsertScraperFileManifestEntry(NewRecord))
+                _log(logattribute::WARNING, "DownloadProjectHostFiles", "Manifest entry already exists for " + NewRecord.hash.ToString() + " " + host_file_name);
+            else
+                _log(logattribute::INFO, "DownloadProjectHostFiles", "Created manifest entry for " + NewRecord.hash.ToString() + " " + host_file_name);
+
+            // The below is not an ideal implementation, because the entire map is going to be written out to disk each time.
+            // The manifest file is actually very small though, and this primitive implementation will suffice. I could
+            // put it up in the while loop above, but then there is a much higher risk that the manifest file could be out of
+            // sync if the wallet is ended during the middle of pulling the files.
+            _log(logattribute::INFO, "ProcessProjectRacFileByCPID", "Persisting manifest entry to disk.");
```

Comment:
Minor copy/paste typo of the function name for the log tag (and below).

Reply:
Ha... eagle eye. When I did the cut and paste from the other function I think I forgot to change the name.

Reply:
I only found it when I <kbd>ctrl-f</kbd>'d for the actual function definition. Thought it was worth mentioning to avoid confusion while looking at the log 🙂

Reply:
Fixed in both (actually 2 x 3) places.